src/anemoi/inference/outputs/assign_mask.py

```diff
@@ -0,0 +1,138 @@
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+import logging
+from typing import Optional
+
+import numpy as np
+
+from anemoi.inference.config import Configuration
+from anemoi.inference.context import Context
+from anemoi.inference.types import State
+
+from ..output import ForwardOutput
+from . import output_registry
+
+LOG = logging.getLogger(__name__)
+
+
+@output_registry.register("assign_mask")
+class AssignMask(ForwardOutput):
+    """Assign the current output to larger area using a mask.
+
+    This operation can be seen as the opposite of "extract_mask".
+    Instead of extracting a smaller area from a larger one,
+    it assigns the current output to a larger area using a mask.
+    This is useful when you want to restore the original state of the model
+    after applying a mask to it. The portion of the state that is not
+    covered by the mask will be set to a fill value (NaN by default).
+
+    Parameters
+    ----------
+    context : dict
+        The context dictionary.
+    output : dict
+        The output configuration dictionary.
+    mask : str
+        The mask supporting array name.
+    fill_value : float, optional
+        The fill value to use for the masked area, by default np.nan.
+    output_frequency : int, optional
+        The frequency of output, by default None.
+    write_initial_state : bool, optional
+        Whether to write the initial state, by default None.
+    """
+
+    def __init__(
+        self,
+        context: Context,
+        *,
+        output: Configuration,
+        mask: str,
+        fill_value: float = np.nan,
+        output_frequency: Optional[int] = None,
+        write_initial_state: Optional[bool] = None,
+    ) -> None:
+        super().__init__(context, output, output_frequency=output_frequency, write_initial_state=write_initial_state)
+
+        if mask not in self.checkpoint.supporting_arrays:
+            raise ValueError(f"Assignment mask '{mask}' not found in supporting arrays.")
+
+        self.mask = self.checkpoint.load_supporting_array(mask)
+        self.fill_value = fill_value
+
+    def __repr__(self) -> str:
+        """Return a string representation of the ExtractLamOutput object."""
+        return f"AssignMask({self.output})"
+
+    def write_initial_state(self, state: State) -> None:
+        """Write the initial step of the state.
+
+        Parameters
+        ----------
+        state : State
+            The state dictionary.
+        """
+        self.output.write_initial_state(self.assign_mask(state))
+
+    def write_step(self, state: State) -> None:
+        """Write a step of the state.
+
+        Parameters
+        ----------
+        state : State
+            The state dictionary.
+        """
+        self.output.write_state(self.assign_mask(state))
+
+    def assign_mask(self, state: State) -> State:
```

Comment:
Related to the above comment: if you rename `assign_mask` to `modify_state` , you should be able to use the super class' implementation of `write_initial_state` and `write_step` ?